### DISCUSSION TITLE
migration steps to upgrade from 3.5.3 to 4.0.0rc1 (docker version)

### DISCUSSION THREAD

**Discussion by GuidoDr** [mastodon/mastodon#19527] [General]
I would like to upgrade our existing installation (v3.5.3 as a docker installation) to 4.0.0rc1.
I tried it with just replacing the tag "latest" to "edge" inside the docker-compose.yml.
The server did start up and did not crash but due to the new role model I am not able to log in.

I do see in the log data that there are a couple of errors, but actually only two different errors:

```PG::UndefinedTable: ERROR:  relation "status_trends" does not exist```
```PG::UndefinedTable: ERROR:  relation "user_roles" does not exist```

So a couple of changes inside the database would probably be needed to get the database to the new database scheme.

Our installation is a small installation and does only have a couple of users (only 5 active users including my admin login).
So I could test the edge version without doing some harms to others.
I like testing of alpha and beta versions, I do this for some other software too.

**Comment by GuidoDr** [comment]
I found it ;-)

BEFORE the migration do run:
`docker-compose run --rm -e SKIP_POST_DEPLOYMENT_MIGRATIONS=true web rails db:migrate`

and AFTER the migration do run:
`docker-compose run --rm web rails db:migrate`

It's running fine now.

**Reply by ggilestro** [reply]
Is this the only manual intervention that is needed? Is there a documentation page guiding the migration?

**Reply by GuidoDr** [reply]
yes that's all what is needed, besides the change in the docker-compose.yml where the image must be set to edge
`image: tootsuite/mastodon:edge`
